src/Controller/EventController.php

```diff
@@ -81,6 +84,37 @@ public function searchEvent(Request $request, EventRepository $eventRepository)
             ];
     }
 
+    /**
+     * @Route("/event/search-dynamic", name="search_events_dynamic")
+     * @Template("event/event_search_dynamic.html.twig")
+     */
+    public function dynamicSearchEvent(Request $request, EventRepository $eventRepository, EventTypeRepository $typeRepository, GenreRepository $genreRepository, SubgenreRepository $subgenreRepository)
+    {
+        $params = $request->query->all();
+//        no validatation on relevant params, just on presence
+        $showResult = boolval(count($params)>0);
+//        no validation params on format, period sanity
+        $params['type'] = $params['type'] ?? [];
+        $params['genre'] = $params['genre'] ?? [];
+        $params['subgenre'] = $params['subgenre'] ?? [];
+        $params['periodStart'] = $params['periodStart'] ?? substr($eventRepository->getMinDate($params),0,10);
+        $params['periodEnd'] = $params['periodEnd'] ?? substr($eventRepository->getMaxDate($params),0,10);
+
+        $start = substr($eventRepository->getMinDate($params),0,10);
+        $end = substr($eventRepository->getMaxDate($params),0,10);
+
+            return [
+                'typeOptions' => $typeRepository->getDynamicFilterOptions($params),
+                'genreOptions' => $genreRepository->getDynamicFilterOptions($params),
+                'subgenreOptions' => $subgenreRepository->getDynamicFilterOptions($params),
```

Comment:
@daria-sieroshtan 
this is a hell lot of duplicate queries.
I don't ask you to **do** this, please **think** how you can reduce DB load